src/odemis/dataio/tiff.py

```diff
@@ -3133,3 +3142,101 @@ def _convert_tescan_to_odemis_metadata(metadata: bytes) -> dict:
+
+    # beam metadata
+    try:
+        beam_type = omd["image"]["beam_type"]
+        try:
+            md[model.MD_DWELL_TIME] = float(omd["image"]["dwell_time"])
+        except KeyError as e:
+            logging.warning(f"Failed to parse OpenFIBSEM metadata: {model.MD_DWELL_TIME} - {e}")
+        try:
+            # MD_POS should match image dimensions, e.g. only XY for 2D, XYZ for 3D, etc.
+            md[model.MD_POS] = (float(omd["microscope_state"]["stage_position"]["x"]),
+                                float(omd["microscope_state"]["stage_position"]["y"])
+                                )
+        except KeyError as e:
+            logging.warning(f"Failed to parse OpenFIBSEM metadata: {model.MD_POS} - {e}")
+
+        try:   
+            acq_map =  {
+                "ELECTRON": model.MD_AT_EM,
+                "ION": model.MD_AT_FIB,
+            }
+            md[model.MD_ACQ_TYPE] = acq_map[beam_type]
+
+        except KeyError as e:
+            logging.warning(f"Failed to parse OpenFIBSEM metadata: {model.MD_ACQ_TYPE} - {e}")
+
+        beam_type_settings = "electron_beam" if beam_type == "ELECTRON" else "ion_beam"
+        beam_settings = omd["microscope_state"][beam_type_settings]
+            
+        try:
+            md[model.MD_ROTATION] = math.radians(float(beam_settings["scan_rotation"]))
+        except KeyError as e:
+            logging.warning(f"Failed to parse OpenFIBSEM metadata: {model.MD_ROTATION} - {e}")
+        
+        try:
+            md[model.MD_EBEAM_VOLTAGE] = float(beam_settings["voltage"])
+        except KeyError as e:
+            logging.warning(f"Failed to parse OpenFIBSEM metadata: {model.MD_EBEAM_VOLTAGE} - {e}")
+        
+        try:    
+            md[model.MD_EBEAM_CURRENT] = float(beam_settings["beam_current"]) 
+        except KeyError as e:
+            logging.warning(f"Failed to parse OpenFIBSEM metadata: {model.MD_EBEAM_CURRENT} - {e}")
+
+        # beam spot size: Not Available
+        try:
+            # detector
+            beam_detector = "electron_detector" if beam_type == "ELECTRON" else "ion_detector"
+            detector_type = omd["microscope_state"][beam_detector]["type"]
+            md[model.MD_DET_TYPE] = (model.MD_DT_NORMAL 
+                                        if detector_type in ["ETD", "TLD", "E-T", "SE"] 
+                                        else model.MD_DT_INTEGRATING)
+        except KeyError as e:
+            logging.warning(f"Failed to parse OpenFIBSEM metadata: {model.MD_DET_TYPE} - {e}")
+
+    except Exception as e:
+        logging.warning(f"Failed to parse OpenFIBSEM metadata: {e}", exc_info=True)
+
+    # if the md is still empty, something else was using ImageDescription tag, 
+    # so we raise an error so external metadata is not updated
+    if not md: 
+        raise ValueError("ImageDescription tag contained data, but no OpenFIBSEM metadata found... falling back to previous metadata...")
```

Comment:
I don't really get that. Sure, it's a sign that it wasn't OpenFIBSEM metadata. But at this point, it's not a big deal, as updating a dict with an empty dict should have no effect. Or am I over looking an issue?

Reply:
You're right that updating an empty dict wont do anything, but I think it is better to explicitly raise the error, rather than silently update with nothing. At the very least it will make debugging easier, if the error is explicit that their was no metadata content